test/units/playbook/test_collectionsearch.py

```diff
@@ -0,0 +1,42 @@
+# (c) 2020 Ansible Project
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from units.compat import unittest
+from ansible.playbook.collectionsearch import CollectionSearch
+from units.compat.mock import patch
+
+
+class TestCollectionSearch(unittest.TestCase):
+
+    @patch('ansible.utils.display.Display.display')
+    def test_collection_static_warning(self, mock_display):
+        '''
+        Test that collection name is not templated and a warning message is
+        generated for the referenced name.
+        '''
+        cs = CollectionSearch()
+        self.assertIn(
+            'foo.{{bar}}',
+            cs._load_collections(None, ['foo.{{bar}}'])
+        )
+        self.assertEqual(mock_display.call_count, 1)
+        self.assertIn(
+            '[WARNING]: "collections" is not templatable, but we found: foo.{{bar}}',
+            mock_display.call_args[0][0])
```

Comment:
I know you've been advised to use `pytest-mock` and [`monkeypatch`](https://docs.pytest.org/en/latest/monkeypatch.html) fixtures but turns out you don't actually need these.

Pytest already captures I/O file descriptors. So what you need is to use what's recorded. This way it's closer to e2e and you don't do unnecessary patching of the logic.

There's [a fixture for capturing warnings `recwarn`](https://docs.pytest.org/en/latest/warnings.html) but it's not what we need here.

Instead, we need [`capsys`](https://docs.pytest.org/en/latest/capture.html#accessing-captured-output-from-a-test-function) (or one of its variations).

And in the end, this test would look like:
```suggestion
from ansible.playbook.collectionsearch import CollectionSearch

import pytest


def test_collection_static_warning(capsys):
    """Test that collection name is not templated.

    Also, make sure that users see the warning message for the referenced name.
    """
    cs = CollectionSearch()
    assert 'foo.{{bar}}' in cs._load_collections(None, ['foo.{{bar}}'])

    std_out, std_err = capsys.readouterr()
    assert '[WARNING]: "collections" is not templatable, but we found: foo.{{bar}}\n' == std_err
    assert '' == std_out
```
_(DISCLAIMER: this is coded in browser, you can accept it by clicking the UI button and see if I haven't made any typos)_